File: 3xp1.js
function threexp1(x) {
    while (x != 1) {
        if (x % 2 == 0) {
            x = x / 2;
        } else {
            x = 3 * x + 1;
        }
        console.log(x);
    }
    console.log("OK");
}

threexp1(27);

File: 3xp1.cpp
#include <iostream>

auto threexp1(int x) -> int {
    while (x != 1) {
        x = x % 2 == 0 ? x / 2 : 3 * x + 1;
        std::cout << x << std::endl;
    }
    std::cout << "OK" << std::endl;
    return 0;
}

auto main() -> int {
    return threexp1(27);
}
